File: Day-2/DynamicArray.js
const numArrays = 9;
const arrays = new Array(numArrays).fill().map(() => []);

let a = 0;

function array(element) {
  const index = a % numArrays;
  arrays[index].push(element);
  a++;
}

array('A');
array('B');
array('C');
array('D');
array('E');
array('F');

for (let i = 0; i < numArrays; i++) {
  console.log(arrays[i]);
}

File: Day-2/DynamicArray.rb
num_arrays = 3
arrays = Array.new(num_arrays) { [] }

a = 0

def array(element, arrays, index)
  arrays[index] << element
end

['A', 'B', 'C', 'D', 'E', 'F'].each do |element|
  index = a % num_arrays
  array(element, arrays, index)
  a += 1
end

arrays.each do |array|
  p array
end
